lib/cc-pico.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2019-2021 Fastly, Janardhan Iyengar, Kazuho Oku
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+#include "quicly/cc.h"
+#include "quicly.h"
+
+#define QUICLY_MIN_CWND 2
+#define QUICLY_PICO_BETA 0.5
+#define QUICLY_PICO_RECOVERY_PERIOD 1000 /* in milliseconds */
```

Comment:
RECOVERY is semantically overloaded with the roundtrip after loss during which cwnd does not decrease again ... I can't think of a good name, but we should think of something else. Maybe SAWTOOTH_PERIOD ? Or EQUILIBRIUM_CYCLE_PERIOD ?

Reply:
As discussed, I've updated the code to use QUICLY_PICO_MULT_HZ, with the unit being Hertz.

Reply:
What's `MULT` stand for?

Reply:
"multiplicative"

Reply:
This PR has been merged, but please let me know if you have suggestions regarding the name.

Reply:
Works for me!